invenio_app_rdm/records_ui/views/records.py

```diff
@@ -279,7 +279,20 @@ def not_found_error(error):
 
 def record_tombstone_error(error):
     """Tombstone page."""
-    return render_template("invenio_app_rdm/records/tombstone.html"), 410
+    # the RecordDeletedError will have the following properties,
+    # while the PIDDeletedError won't
+    record = getattr(error, "result_item", None)
+    raw_record = getattr(error, "record", None)
+
+    if raw_record and raw_record.tombstone and not raw_record.tombstone.is_visible:
+        return not_found_error(error)
```

Comment:
```suggestion
        raise not_found_error(error)
```

Reply:
just as a bit of context:
the `not_found_error` i'm using here is the function defined a few lines above, which renders the 404 template and not an exception.
this view function (`record_tombstone_error`) is called when the error handler catches a `RecordDeletedException` already, so raising another exception here to go back to the error handler might be a bit complicated.

Reply:
I think in general it would be nicer to handle it with the handler, not sure what others think

then I would propose to change the name of the function, otherwise it is suggesting we deal with error/exception

Reply:
I agree with the naming, it really sounds like an error more than a handler function.
I didn't touch any of the naming though, in order to avoid unnecessary churn and stay consistent with what's already there.

About the error raising from the error handler, I personally feel that this would introduce a layer of indirection that makes the logic harder to follow.

Reply:
Further, exceptions raised in the error handlers don't seem to go back up to where the errors are routed to their registered handlers – I just tried raising a `KeyError("kekw")` instead of `return not_found_error(error)`, with the following result:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/flask/app.py", line 2552, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 187, in __call__
    return self.app(environ, start_response)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/werkzeug/middleware/dispatcher.py", line 78, in __call__
    return app(environ, start_response)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/flask/app.py", line 2532, in wsgi_app
    response = self.handle_exception(e)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/flask/app.py", line 1676, in handle_user_exception
    return self.ensure_sync(handler)(e)
  File "/home/mmoser/rdm/modules/invenio-app-rdm/invenio_app_rdm/records_ui/views/records.py", line 293, in record_tombstone_error
    raise KeyError("kekw")
KeyError: 'kekw'
```

Even though there is an [error handler registered for `KeyError`s](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/views/__init__.py#L140).